architecture-decision-records/0011-base25-encoding-for-consignment-reference.md

```diff
@@ -0,0 +1,88 @@
+
+### Option 2: PostgreSQL Sequence
+
+A sequence is a database object that can generate a big-int (8-bit) and be associated with specific tables. Using a sequence is slightly more manual, but  allows for many customisable options (min/max value, starting point, cycle/no-cycle).
+
+#### Advantages
+
+* We can state specifically that we don't want the value to cycle when max-value is reached (extremely unlikely to happen)
+* We can start from a specific value, which would be useful if tables/db is dropped and sequence needs to be initiated again.
+* Can be added to existing tables
+
+#### Disadvantages
+
+* A slightly more manual set up, but behaves the same as serial once initiated.
+
+## Options considered for encoding
+
+### Option 1: DRI code
+
+We could manually encode these ID numbers to base25 utilising code that is similar to that written for the [DRI](https://nationalarchivesuk.sharepoint.com/sites/DA_DPT/Systems/Forms/AllItems.aspx?id=%2Fsites%2FDA%5FDPT%2FSystems%2FDigital%20Records%20Infrastructure%2FDocumentation%2FGenerated%20Catalogue%20References%20%2D%20Draft%20v1%2E0%2Epdf&parent=%2Fsites%2FDA%5FDPT%2FSystems%2FDigital%20Records%20Infrastructure%2FDocumentation&RootFolder=%2Fsites%2FDA%5FDPT%2FSystems%2FDigital%20Records%20Infrastructure%2FDocumentation&FolderCTID=0x012000E56AFAD10E754045898F8F352035CA1F00E09F204D3696E04193FD4FFBF9EA85B7) project in 2014. 
+This code could be copy and pasted within the TDR consignment API to be used as we see fit. We would explicitly define the required list of available alpha-numeric values and create our own encode/decode functions according to this 'alphabet'.
+
+#### Advantages
+
+* Customisable to what we want - especially if our requirements change
+* Easy to read for humans and see what it does at first glance
+
+#### Disadvantages
+
+* Based on code that is slightly older
+* Could potentially be bulky to incorporate into the consignment API
+
+### Option 2: [Omega Catalog Identifier tools (Scala)](https://github.com/nationalarchives/oci-tools-scala)
+
+This [library](https://search.maven.org/artifact/uk.gov.nationalarchives.oci/oci-tools-scala_2.13/0.2.0/jar) can encode to any specified base*N*, and decode back to base10. The Omega library has some pre-specified 'alphabets' included within, one of which is GCRb25, meaning an alphabet does not need to be explicitly created to utilise this code.
+The library is a little more complex than utilising the 2014 DRI code, and more than one function needs to be called to encode from an integer to the correct code, but working with this is relatively straight-forward.
+
+#### Advantages
+
+* Can encode to any base*N* not just base25/GCRb25
+* Already exists as a tool, meaning encoding functionality does not need to be created by TDR team
+* Error handling built in
+
+#### Disadvantages
+
+* Open source - the aim of the project may change and affect the way the TDR team can use the library
+* Less customisable - more generalised
+
+## Decisions
```

Comment:
Can you reiterate in this section that the format will be `TDR-<year>-<base25-ID>`? Just so it's clear that this is part of the decision we're recording here.